lamindb/curators/__init__.py

```diff
@@ -1777,15 +1780,19 @@ def __init__(
         *,
         sample_metadata_key: str | None = "sample",
     ) -> None:
-        if sources is None:
-            sources = {}
-        self._sources = sources
-        if exclude is None:
-            exclude = {}
-        self._exclude = exclude
-        self._sdata: SpatialData = sdata
+        super().__init__(
+            dataset=sdata,
+            categoricals={},
+            sources=sources,
+            organism=organism,
+            exclude=exclude,
+        )
+        if isinstance(sdata, Artifact):
+            # TODO: load() doesn't yet work
+            self._sdata = sdata.load()
```

Comment:
This is also problematic because this spatial data object might be huge.

Reply:
I know, but there isn't really another way get the in-memory object. If we allow user to pass both the object and artifact, that will be too dangerous because they might not match.